File: app/src/main/java/pl/derjack/yabreakthrough/GameViewModel.java
package pl.derjack.yabreakthrough;

import android.util.Log;

import androidx.annotation.Nullable;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

import pl.derjack.yabreakthrough.game.Action;
import pl.derjack.yabreakthrough.game.Game;

public class GameViewModel extends ViewModel {

    @Nullable
    public Game game;
    @Nullable
    public CpuWorker cpuWorker;
    @Nullable
    private SingleLiveEvent<Action> lastAction;

    public MutableLiveData<Action> getLastAction() {
        if (lastAction == null) {
            lastAction = new SingleLiveEvent<>();
        }
        return lastAction;
    }

    @Override
    protected void onCleared() {
        Log.d("GameViewModel","onCleared");
    }
}

File: app/src/main/java/pl/derjack/yabreakthrough/GameViewModel.kt
package pl.derjack.yabreakthrough

import android.util.Log
import androidx.lifecycle.ViewModel
import pl.derjack.yabreakthrough.game.Game
import pl.derjack.yabreakthrough.CpuWorker
import pl.derjack.yabreakthrough.SingleLiveEvent
import androidx.lifecycle.MutableLiveData
import pl.derjack.yabreakthrough.game.Action

class GameViewModel : ViewModel() {
    var game: Game? = null
    var cpuWorker: CpuWorker? = null
    val lastAction: SingleLiveEvent<Action> by lazy { SingleLiveEvent() }

    override fun onCleared() {
        Log.d("GameViewModel", "onCleared")
    }
}